### PR TITLE
[REVIEW] Feature/GitHub api

### PR DESCRIPTION
**Pull request by tehmou**
- Major update
- Change to use GitHub repositories API
- Add ContentProvider support
- Add Retrolambda

### CODE REVIEW

File: app/src/main/java/com/tehmou/rxbookapp/data/ContentProviderStoreBase.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        }
+    }
+
+    protected T query(U id) {
+        return query(getUriForId(id));
+    }
+
+    protected T query(Uri uri) {
+        Cursor cursor = contentResolver.query(uri, SerializedJsonContract.PROJECTION, null, null, null);
+        T gitHubRepository = null;
```
Comment (jaggernod) [inline app/src/main/java/com/tehmou/rxbookapp/data/ContentProviderStoreBase.java]: The class is generic, would be there a way to have the contract more abstract. Somehow to remove any "gitHub" specific stuff

File: app/src/main/java/com/tehmou/rxbookapp/data/GitHubRepositorySearchStore.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    @Override
+    protected String getIdFor(GitHubRepositorySearch item) {
+        return item.getSearch();
+    }
+
+    @Override
+    protected Uri getContentUri() {
+        return GitHubRepositorySearchContract.CONTENT_URI;
+    }
+}
```
Comment (jaggernod) [inline app/src/main/java/com/tehmou/rxbookapp/data/GitHubRepositorySearchStore.java]: No new line at the end of the file

File: app/src/main/java/com/tehmou/rxbookapp/utils/TextWatcherObservable.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
+package com.tehmou.rxbookapp.utils;
```
Comment (jaggernod) [inline app/src/main/java/com/tehmou/rxbookapp/utils/TextWatcherObservable.java]: Why not using RxAndroid stuff? https://github.com/ReactiveX/RxAndroid/blob/0.x/rxandroid/src/main/java/rx/android/widget/OnTextChangeEvent.java

File: app/src/main/java/com/tehmou/rxbookapp/viewmodels/RepositoriesViewModel.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+    public RepositoriesViewModel() {
+        RxBookApp.getInstance().getGraph().inject(this);
+        Log.v(TAG, "RepositoriesViewModel");
+    }
+
+    public void subscribeToDataStore() {
+        Log.v(TAG, "subscribeToDataStore");
+
+        compositeSubscription.add(
```
Comment (jaggernod) [inline app/src/main/java/com/tehmou/rxbookapp/viewmodels/RepositoriesViewModel.java]: You are subscribing and unsubscribing several times from it. And CompositeSubscription has to be recreated then. 

http://reactivex.io/RxJava/javadoc/rx/subscriptions/CompositeSubscription.html#clear()

File: app/src/main/java/com/tehmou/rxbookapp/data/ContentProviderStoreBase.java (hunk 1 above)
Comment (tehmou) [inline app/src/main/java/com/tehmou/rxbookapp/data/ContentProviderStoreBase.java]: Yeah, a good idea. I think I forgot the names there when refactoring.

**Pull request merged**